pulsar4s-core/src/main/scala/com/sksamuel/pulsar4s/Producer.scala

```diff
@@ -4,10 +4,36 @@ import java.io.Closeable
 
 import com.sksamuel.exts.Logging
 import org.apache.pulsar.client.api.{ProducerStats, TypedMessageBuilder}
+import org.apache.pulsar.client.api.transaction.Transaction
 
 import scala.util.Try
 
-trait Producer[T] extends Closeable with Logging {
+trait TransactionalProducerOps[T] {
+
+  /**
+    * Asynchronously sends a message of type T, returning an effect
+    * which will be completed with the [[MessageId]] once the message
+    * is acknowledged by the Pulsar broker.
+    *
+    * This method can be used when you have no need to set the
+    * other properties of a message, such as the event time, key,
+    * headers and so on. The producer will generate an appropriate
+    * Pulsar [[ProducerMessage]] with this t set as the value.
+    */
+  def sendAsync[F[_] : AsyncHandler](t: T): F[MessageId]
```

Comment:
I noticed that Pulsar supports both `send` and `sendAsync` with transactions, while it only supports `acknowledgeAsync`, but it seems odd that you'd want to use an async method one place but not in another, so I think I'll omit that from the transactional API.